src/fsharp/CheckExpressions.fs

```diff
@@ -6739,27 +6739,35 @@ and TcInterpolatedStringExpr cenv overallTy env m tpenv (parts: SynInterpolatedS
 
         UnifyTypes cenv env m printerTupleTy printerTupleTyRequired
 
-        // Type check the expressions filling the holes
-        let flexes = argTys |> List.map (fun _ -> false)
-        let fillExprs, tpenv = TcExprs cenv env m tpenv flexes argTys synFillExprs
+        if List.isEmpty synFillExprs then
+            let str = mkString g m printfFormatString
```

Comment:
In theory this changes the quotation form for an interpolated string.  I don't mind that change, however could you add a test in `tests/fsharp/core/quotes/test.fsx` or elsewhere to test for that and pin down the new form please?

e.g. check results of `<@ $"abc" @>`  and you may as well add a test for `<@ $"abc {1} def" @>` while you're at it.